feat(experience): header category navigation

tobi-or-not-tobi opened the pull request:
The header navigation contains some of the categories available in the assortment, which differ from b2c to b2b. The category is build with standard components, such compositions and link components, and the layout is provided by a new navigation layout. The navigation layout supports both horizontal or vertical layouts.

The navigation layouts adds a number of (opinionated) CSS variables that are used in the styles of the descendants content link components:
- `--oryx-link-icon-size`
- `--oryx-link-width`
- `--oryx-link-color`
- `--oryx-link-decoration`
- `--oryx-link-padding`
- `--oryx-link-hover-shadow`
- `--oryx-link-active-shadow`
- `--oryx-link-current-shadow`

closes: [HRZ-2519](https://spryker.atlassian.net/browse/HRZ-2519)

[HRZ-2519]: https://spryker.atlassian.net/browse/HRZ-2519?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

tobi-or-not-tobi committed:
artificial commit

tobi-or-not-tobi committed:
chore: Navigation layout plugin implementation (#883)

Navigation layout plugin and shadow layout property plugin. 

closes: [HRZ-90383](https://spryker.atlassian.net/browse/HRZ-90383)
[HRZ-90391](https://spryker.atlassian.net/browse/HRZ-90391)
[HRZ-90390](https://spryker.atlassian.net/browse/HRZ-90390)

[HRZ-90383]:
https://spryker.atlassian.net/browse/HRZ-90383?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[HRZ-90391]:
https://spryker.atlassian.net/browse/HRZ-90391?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[HRZ-90390]:
https://spryker.atlassian.net/browse/HRZ-90390?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

tobi-or-not-tobi committed:
Merge branch 'development' into feature/HRZ-2519-header-navigation

tobi-or-not-tobi committed:
fix SSR

tobi-or-not-tobi committed:
ssr improvements

tobi-or-not-tobi committed:
Merge branch 'development' into feature/HRZ-2519-header-navigation

tobi-or-not-tobi committed:
refactor: category links refactor (#893)

we've changed the styles so that navigation styles can peers through the
shadow DOM and affect the CSS of content links whether they're direct
children or nested in other components (e.g. a category-list).

Sinucid commented on libs/domain/search/box/box.component.spec.ts:
```diff
@@ -93,7 +93,9 @@ describe('SearchBoxComponent', () => {
       ],
     });
-    linkService = testInjector.inject(LinkService) as MockSemanticLinkService;
+    linkService = testInjector.inject(
+      LinkService
+    ) as unknown as MockSemanticLinkService;
```
```suggestion
    linkService = testInjector.inject<MockSemanticLinkService>(
      LinkService
    );
```

tolerants commented on libs/platform/experience/src/services/layout/plugins/types/navigation/navigation-layout.plugin.ts:
```diff
@@ -0,0 +1,30 @@
+  getStyles(data: LayoutPluginOptionsParams): Observable<LayoutStyles> {
+    const { options } = data;
+
+    return ssrAwaiter(
```
Is it a right usage of `ssrAwaiter`? I mean, why should layout know something about ssr?

tobi-or-not-tobi replied:
yup, this is standard approach we took in the plugins to ensure we can lazy load styles.

tobi-or-not-tobi committed:
fix

tobi-or-not-tobi committed:
Merge branch 'development' into feature/HRZ-2519-header-navigation

tobi-or-not-tobi committed:
Merge branch 'development' into feature/HRZ-2519-header-navigation

tobi-or-not-tobi committed:
chore: category navigation e2e (#892)

Test whether the category navigation contains a link to all products and
whether the navigation links work.

closes: [HRZ-90392](https://spryker.atlassian.net/browse/HRZ-90392)


[HRZ-90392]:
https://spryker.atlassian.net/browse/HRZ-90392?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

tobi-or-not-tobi committed:
Merge branch 'development' into feature/HRZ-2519-header-navigation

tobi-or-not-tobi committed:
Merge branch 'development' into feature/HRZ-2519-header-navigation

tobi-or-not-tobi committed:
Merge branch 'development' into feature/HRZ-2519-header-navigation
